[Refactor] - Pre-commit hooks

barrymun opened the pull request:
Thank you for your contribution to the ClintonCAT repo.
Before submitting this PR, please make sure:

- [x] The target of this PR is the `main` branch.
- [x] No commits are missing from forked base branch (e.g. modified main branch instead of dev)

### Changes
- Install `husky` pre-commit hooks so linting and tests are executed before any changes are committed (any other commits to be executed pre-commit can be added to the `.husky/pre-commit` file).
- Adding prettier to the eslint config so linting will pick up prettier errors and prettier errors will be shown in the IDE/editor.
- Change prettier line endings to auto to cater for windows and mac devs.

WayneKeenan commented:
please can you yank the pages_db*.json files, they are autogenerated so I'll fix the the generation script. 
Thank you

Producdevity commented on package.json:
```diff
@@ -5,12 +5,13 @@
         "test": "jest",
-        "lint": "eslint \"src/**/*.{ts,tsx}\"",
-        "format": "prettier --write \"src/**/*.{ts,tsx,js,jsx,json,md}\"",
+        "lint": "eslint \"**/*.{ts,tsx,js,jsx,json,md}\"",
```
```suggestion
        "typecheck": "tsc --noEmit",
        "lint": "eslint \"**/*.{ts,tsx,js,jsx,json,md}\"",
```

Producdevity commented on .husky/pre-commit:
```diff
@@ -0,0 +1,5 @@
+npm run format
+npm run lint
```
```suggestion
npm run lint
npm run typecheck
```

Producdevity commented:
I added a couple suggested code changes:
- adding npm run typecheck to package.json and git hook
- excluding `.idea` from prettier

barrymun commented:
> please can you yank the pages_db*.json files, they are autogenerated so I'll fix the the generation script. Thank you

These changes have been reverted, and the `data` directory is ignored when running `format`.

barrymun commented:
> I added a couple suggested code changes:
> 
> * adding npm run typecheck to package.json and git hook
> * excluding `.idea` from prettier

Thanks, I've implemented these changes. Let me know if you see any other problems.

WayneKeenan commented:
thanks for your time on this.
